rednotebook/journal.py

```diff
@@ -555,6 +555,42 @@ def search(self, text, tags):
             results.append(day.search(text, tags))
         return results
 
+    def replace_all(self, old, new):
+        """Replaces all strings matching 'old' with the 'new' string"""
+        total_replacements = 0
+
+        # Replacement for the current frame.
+        replacements = self.day.replace_all(old, new)
+        if replacements > 0:
+            self.month.edited = True
+            self.frame.set_day_text(self.day.text)
+
+            # Update the preview if linux
+            if self.frame.preview_mode and self.frame.html_editor.internal:
+                self.frame.html_editor.show_day(self.day)
+
+            total_replacements += replacements
+
+        # Replacement for all other days
```

Comment:
This updates the current day again, right? Can we do this loop first, and then do the special code for showing the changes to the current day?